lib/disk/modules/VMWareSparseDisk.rb

```diff
@@ -139,16 +189,67 @@ def d_size
 
   private
 
+  def initialize_without_markers
+    #
+    # Seek to start of the grain directory.
+    #
+    @vmwareSparseDisk_file.seek(@sparseHeader.gdOffset * blockSize, IO::SEEK_SET)
+
+    #
+    # Read the first grain directory entry to get the offset to the start of
+    # the grain tables.
+    #
+    buf = @vmwareSparseDisk_file.read(BYTES_PER_SECTOR)
+    @grainDirectory = []
+
+    (0..(GDES_PER_GD - 1)).each do |index|
```

Comment:
Prefer `(0...GDES_PER_GD)`  (3 dots means exclusive end)

Alternately

```
GDES_PER_GD.times do |index|
```

Reply:
I like the `times` approach.  Will change to that.